aten/src/ATen/native/mps/operations/Distributions.mm

```diff
@@ -341,6 +342,44 @@ Tensor normal_mps(const Tensor& mean, const Tensor& std, c10::optional<Generator
                                       "exponential_mps_:" + std::to_string(lambda), random_op_block);
 }
 
+Tensor& randperm_out_mps(int64_t n, c10::optional<Generator> generator, Tensor& result) {
+  if (!MPSDevice::getInstance()->macOS_13_0()) {
+    TORCH_WARN_ONCE("MPS: randperm op is supported natively starting from macOS 13.0. ",
+                    "Falling back on CPU. This may have performace implications.");
+
+    result = result.to("cpu");
+    result = at::randperm_out(result, n).to("mps");
+    return result;
+  }
+
+  using namespace mps;
```

Comment:
You've already added `mps::` to places you needed, so why you need this?

Reply:
I think I added that just for getTensorsStringKey({result}) 😄 . I'll update